math/vecops/test/vecops_rvec.cxx

```diff
@@ -1347,3 +1323,470 @@ TEST(VecOps, NoExceptionSafety)
+   EXPECT_TRUE(IsAdopting(vadopt1));
+   EXPECT_TRUE(IsAdopting(vadopt2));
+
+   test_swap(vadopt1, vadopt3); // left vector has bigger size
+
+   CheckEqual(vadopt1, v3);
+   CheckEqual(vadopt3, v2);
+   EXPECT_EQ(&vadopt1[0], &v3[0]);
+   EXPECT_EQ(&vadopt3[0], &v2[0]);
+   EXPECT_FALSE(IsSmall(vadopt1));
+   EXPECT_FALSE(IsSmall(vadopt3));
+   EXPECT_TRUE(IsAdopting(vadopt1));
+   EXPECT_TRUE(IsAdopting(vadopt3));
+
+   test_swap(vadopt1, vadopt3); // left vector has smaller size
+
+   CheckEqual(vadopt1, v2);
+   CheckEqual(vadopt3, v3);
+   EXPECT_EQ(&vadopt1[0], &v2[0]);
+   EXPECT_EQ(&vadopt3[0], &v3[0]);
+   EXPECT_FALSE(IsSmall(vadopt1));
+   EXPECT_FALSE(IsSmall(vadopt3));
+   EXPECT_TRUE(IsAdopting(vadopt1));
+   EXPECT_TRUE(IsAdopting(vadopt3));
+}
+
+// there is a difference between std::swap and ROOT::VecOps::swap here
+// they both produce the same result, but std::swap might produce 2 regular vectors
+// in cases where ROOT::VecOps::swap produces 1 regular and 1 adopting vector
+// therefore ROOT::VecOps::swap-s are mainly used "alone"
+TEST_P(VecOpsSwap, SmallRegularVectors)
+{
+   RVec<int> fixed_vsmall{1, 2, 3};
+   RVec<int> fixed_vreg1{4, 5, 6};
+   RVec<int> fixed_vreg2{7, 8};
+   RVec<int> fixed_vreg3{9, 10, 11, 12, 13, 14};
+   RVec<int> fixed_vreg4{15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30};
+
+   // need multiple hard copies since after swap of a small and a regular,
+   // there is no fixed policy whether 2 regular vectors are produced or 1 small and 1 regular
+   RVec<int> vsmall1{1, 2, 3};
+   RVec<int> vsmall2{1, 2, 3};
+   RVec<int> vsmall3{1, 2, 3};
+   RVec<int> vsmall4{1, 2, 3};
+   RVec<int> vsmall5{1, 2, 3};
+   RVec<int> vsmall6{1, 2, 3};
+   RVec<int> vsmall7{1, 2, 3};
+
+   RVec<int> vreg1{4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6};
+   vreg1.erase(vreg1.begin() + 3, vreg1.end()); // regular vector of size 3
+   RVec<int> vreg2{7, 8, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6};
+   vreg2.erase(vreg2.begin() + 2, vreg2.end()); // regular vector of size 2
+   RVec<int> vreg20{7, 8, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6};
+   vreg20.erase(vreg20.begin() + 2, vreg20.end()); // regular vector of size 2
+   RVec<int> vreg3{9, 10, 11, 12, 13, 14, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6};
+   vreg3.erase(vreg3.begin() + 6, vreg3.end()); // regular vector of size 6
+   RVec<int> vreg30{9, 10, 11, 12, 13, 14, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6};
+   vreg30.erase(vreg30.begin() + 6, vreg30.end()); // regular vector of size 6
+   RVec<int> vreg4{15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30};
+   // vreg4 is a regular vector that cannot "fit" to small vector
+
+   // verify that initally vectors are not small
+   EXPECT_FALSE(IsSmall(vreg1));
+   EXPECT_FALSE(IsSmall(vreg2));
+   EXPECT_FALSE(IsSmall(vreg3));
+   EXPECT_FALSE(IsSmall(vreg4));
+
+   swap(vsmall1, vreg1); // small <-> regular (same size)
```

Comment:
just noticed. shouldn't this and the calls below be `test_swap`?

Reply:
Actually no. I have a comment on the top of the block (line ~1520). In this case `swap` produces a small and a regular vector, while `std::swap` produces 2 regular vectors. As possible solutions I can:
1. Write explicitly another test for `std::swap`.
2. Disable the check for small or/and regular on the vector that differs.
3. Simply mention on the top of the test block: `therefore ROOT::VecOps::swap-s are mainly used "alone"`.

I am not able to write `if (GetValue()) <expect true> else <expect false>`. I considered it as the "cleanest" solution to just stick to `swap`.